src/Plugins/Reminder.php

```diff
@@ -0,0 +1,357 @@
+                    }
+                    break;
+                case 'reminder':case 'remind':
+                    $parameters = implode(" ", $command->getParameters());
+                    if(!preg_match(self::REMINDER_REGEX, $parameters, $matches)){
+                        return $this->chatClient->postMessage($command->getRoom(), self::USAGE);
+                    }
+
+                    $time = $matches[2] ?? false;
+                    $text = $matches[1] ?? false;
+                    break;
+            }
+
+            if(!isset($time) || !$time): return $this->chatClient->postMessage($command->getRoom(), "Have a look at the time again, yo!"); endif;
+            if(!isset($text) || !$text): return $this->chatClient->postMessage($command->getRoom(), self::USAGE); endif;
+
+            $timestamp = strtotime($time) ?: strtotime("+{$time}"); // false|int
+            if (!$timestamp): return $this->chatClient->postMessage($command->getRoom(), "Have a look at the time again, yo!"); endif;
+
+            $messageId = $command->getId();
+            $key = base_convert($messageId, 10, 36);
+            $value = [
+                'id' => $messageId,
+                'text' => $text,
+                'delay' => $time,
+                'userId' => $command->getUserId(),
+                'username' => $command->getUserName(),
+                'timestamp' => $timestamp
+            ];
+
+            $seconds = $timestamp - time();
+            if ($seconds <= 0): return $this->chatClient->postReply($command, "I guess I'm late: " . $text); endif;
+
+            if( yield $this->storage->set($key, $value, $command->getRoom()) ){
+
+                $watcher = once(function () use ($command, $value, $key) {
+                    yield $this->storage->unset($key, $command->getRoom());
+                    return $this->chatClient->postReply($command, $value['text']);
+                }, $seconds * 1000);
+
+                $this->watchers[] = $watcher;
+                return $this->chatClient->postMessage($command->getRoom(), "Reminder set.");
+            }
+
+            return $this->chatClient->postMessage($command->getRoom(), "Dunno what happened but I couldn't set the reminder.");
+        });
+    }
+
+    private function getAllReminders(Command $command): Promise
+    {
+        return resolve(function() use($command) {
+            $message = "Registered reminders are:";
+
+            $reminders = yield $this->storage->getAll($command->getRoom());
+            if($reminders === []){
+                return $this->chatClient->postMessage($command->getRoom(), "There aren't any scheduled reminders.");
+            }
+
+            $timeouts = [];
+            foreach ($reminders as $item => $value) {
+                $text = $value['text'];
+                $user = $value['username'];
+                $key  = base_convert($value['id'], 10, 36);
```

Comment:
I'll use the `$item` here and rename the variable but, the key has to be a string and the `$command->getId()` returns integer. Before I was adding `":" . $value['id'];` but this seemed like a good option since it also gives a shorter string. Can revert back to adding ":" or any other solution, open to suggestions.